test/new-e2e/tests/agent-platform/common/test_client.go

```diff
@@ -281,3 +282,93 @@ func ReadJournalCtl(t *testing.T, client *TestClient, grepPattern string) string
 	}
 	return journalCtlOutput
 }
+
+// DockerTestClient is a helper to run commands on a docker container for tests
+type DockerTestClient struct {
+	host          *components.RemoteHost
+	containerName string
+}
+
+// NewDockerTestClient creates a client to help write tests that run on a docker container
+func NewDockerTestClient(host *components.RemoteHost, containerName string) DockerTestClient {
+	return DockerTestClient{
+		host:          host,
+		containerName: containerName,
+	}
+}
+
+// RunContainer starts the docker container in the background based on the given image reference
+func (c *DockerTestClient) RunContainer(image string) error {
+	// We run an infinite no-op to keep it alive
+	_, err := c.host.Execute(
+		fmt.Sprintf("docker run -d -e DD_HOSTNAME=docker-test --name '%s' '%s' tail -f /dev/null", c.containerName, image),
+	)
+	return err
+}
+
+// Cleanup force-removes the docker container associated to the client
+func (c *DockerTestClient) Cleanup() error {
+	_, err := c.host.Execute(fmt.Sprintf("docker rm -f '%s'", c.containerName))
+	return err
+}
+
+// Execute runs commands on a Docker remote host
+func (c *DockerTestClient) Execute(command string) (output string, err error) {
+	return c.host.Execute(
+		// Run command on container via docker exec,
+		// wrap in a shell call to achieve similar behavior to remote hosts
```

Comment:
*nit*:  unnecessary comment

Reply:
How is this unnecessary? If anything, it could be improved, but the fact that we use `sh -c <command>` is pretty important to allow the same behavior as the `Execute` function works for remote hosts (i.e. support redirects, pipes and the like).

Reply:
But I mean it is clear as it describe the code below, I will resolve it.